tests/test_data_collection.py

```diff
@@ -1,25 +1,81 @@
+"""
+Test module to determine the correctness of the data_collection.py file.
+
+The test case will take the current repository.
+
+Unless that path variable is changed.
+"""
 import pytest
 from src import data_collection
+from src import json_handler
 
 
 @pytest.mark.xfail
 @pytest.mark.parametrize(
-    "repository_url",
-    [("https://github.com/GatorCogitate/cogitate_tool")],
+    "repository_url", [("https://github.com/GatorCogitate/cogitate_tool")],
 )
-def test_collect_commits_user_email_key(repository_url):
+def test_collect_commits_user_hash(repository_url):
     dict = {}
     assert len(dict) == 0
-    dict = data_collection.collect_commits_user_email_key(repository_url)
+    dict = data_collection.collect_commits_hash(repository_url)
     assert len(dict) != 0
 
 
 @pytest.mark.parametrize(
-    "repository_url",
-    [("https://github.com/GatorCogitate/cogitate_tool")],
+    "repository_url", [("https://github.com/GatorCogitate/cogitate_tool")],
 )
 def test_collect_commits_hash(repository_url):
     list = []
     assert len(list) == 0
     list = data_collection.collect_commits_hash(repository_url)
     assert len(list) != 0
+
+
+def test_raw_data_exists_in_testfile():
+    """Checks the existence of the key RAW_DATA in individual_metrics_testfile."""
+    test_dict = json_handler.get_dict_from_json_file("individual_metrics_testfile")
+    assert "RAW_DATA" in test_dict.keys()
+
+
+def test_calculate_individual_metrics_populates_data():
+    """Checks that the function retruns a populated dictionary."""
+    test_dict = {}
+    # pylint: disable=len-as-condition
+    assert len(test_dict) == 0
+    test_dict = data_collection.calculate_individual_metrics(
+        "individual_metrics_testfile"
+    )
+    assert len(test_dict) != 0
+    assert "INDIVIDUAL_METRICS" in test_dict.keys()
+
+
+def test_get_individual_metrics_accuracy():
+    """Checks that individual_metrics data outputs correct values."""
+    test_dict = data_collection.calculate_individual_metrics(
+        "individual_metrics_testfile"
+    )
+    expected_dict = {
+        "EMAIL": "buchin@allegheny.edu",
+        "COMMITS": 1,
+        "ADDED": 694,
+        "REMOVED": 0,
+        "TOTAL": 694,
+        "MODIFIED": 694,
+        "RATIO": 694,
+        "FILES": ["Pipfile", "Pipfile.lock", "UsingPyGithub.py", "lint.sh", "test.sh"],
+        "FORMAT": [".lock", ".py", ".sh", "Pipfile"],
+    }
+    assert test_dict["INDIVIDUAL_METRICS"]["noorbuchi"] == expected_dict
```

Comment:
I would suggest using something like `tmpdir` in the future to test this kind of functions where it needs to interact with the file systems. `tmpdir` creates a temporary directory for the test case use, that way you don't need to actually have this file in the repo.

Reply:
Great point! we learned about this feature today in class and hopefully, we will rework our test cases to use it in the future.